earth_enterprise/src/common/khxml/khxml.cpp

```diff
@@ -161,18 +220,38 @@ class UsingXMLGuard
 };
 
 static khMutexBase xmlLibLock = KH_MUTEX_BASE_INITIALIZER;
+static khMutexBase checkTermLock = KH_MUTEX_BASE_INITIALIZER;
+static khMutexBase cacheLock = KH_MUTEX_BASE_INITIALIZER;
+
+// only terminate when certain conditions are met
+void  readyForTerm()
```

Comment:
Minor: The name of this function makes it sound like it doesn't actually do the terminate. Maybe change to something like `reinitIfReady`?